app/src/main/java/org/simple/clinic/teleconsultlog/medicinefrequency/MedicineFrequencyBottomSheet.kt

```diff
@@ -0,0 +1,131 @@
+import org.simple.clinic.ClinicApp
+import org.simple.clinic.R
+import org.simple.clinic.mobius.MobiusDelegate
+import org.simple.clinic.teleconsultlog.drugduration.DrugDuration
+import org.simple.clinic.teleconsultlog.medicinefrequency.di.MedicineFrequencyComponent
+import org.simple.clinic.util.unsafeLazy
+import org.simple.clinic.widgets.BottomSheetActivity
+import javax.inject.Inject
+
+class MedicineFrequencyBottomSheet : BottomSheetActivity(), MedicineFrequencySheetUiActions {
+
+  @Inject
+  lateinit var effectHandlerFactory: MedicineFrequencyEffectHandler.Factory
+
+  private lateinit var component: MedicineFrequencyComponent
+
+  companion object {
+    private const val DRUG_DURATION = "drugDuration"
+    private const val MEDICINE_FREQUENCY = "medicineFrequency"
+    private const val SAVED_MEDICINE_FREQUENCY = "savedmedicineFrequency"
+
+    fun intent(
+        context: Context,
+        drugDuration: DrugDuration,
+        medicineFrequency: MedicineFrequencies
+    ): Intent {
+      return Intent(context, MedicineFrequencyBottomSheet::class.java).apply {
+        putExtra(DRUG_DURATION, drugDuration)
+        putExtra(MEDICINE_FREQUENCY, medicineFrequency)
+      }
+    }
+  }
+
+  private val drugDuration by unsafeLazy {
+    intent.getParcelableExtra<DrugDuration>(DRUG_DURATION)!!
+  }
+
+  private val medicineFrequency by unsafeLazy {
+    intent.getSerializableExtra(MEDICINE_FREQUENCY)
+  }
+
+  private val events by unsafeLazy {
+    saveClicks()
+  }
+
+  private val delegate by unsafeLazy {
+    MobiusDelegate.forActivity(
+        events = events.ofType(),
+        defaultModel = MedicineFrequencyModel.create(MedicineFrequencies.BD),
+        init = MedicineFrequencyInit(),
+        update = MedicineFrequencyUpdate(),
+        effectHandler = effectHandlerFactory.create(this).build()
+    )
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.sheet_medicine_frequency)
+    medicineFrequencyTitleTextView.text = getString(R.string.drug_duration_title, drugDuration.name, drugDuration.dosage)
+  }
+
+  private fun saveClicks(): Observable<MedicineFrequencyEvent> {
+    return saveMedicineFrequencyButton
+        .clicks()
+        .map {
+          when {
+            medicineFrequencyOdRadioButton.isChecked -> medicineFrequency == MedicineFrequencies.OD
+            medicineFrequencyBdRadioButton.isChecked -> medicineFrequency == MedicineFrequencies.BD
+            medicineFrequencyTdsRadioButton.isChecked -> medicineFrequency == MedicineFrequencies.TDS
+            else -> medicineFrequency == MedicineFrequencies.QDS
+          }
+          SaveMedicineFrequencyClicked(medicineFrequency as MedicineFrequencies)
+        }
+  }
+
+  override fun setMedicineFrequency(medicineFrequency: MedicineFrequencies) {
+    medicineFrequencyOdRadioButton.isChecked = medicineFrequency == MedicineFrequencies.OD
```

Comment:
While this works, there is a better way to check the radio buttons in `RadioGroup` using [`RadioGroup#check`](https://developer.android.com/reference/android/widget/RadioGroup#check(int)). `RadioGroup` will automatically uncheck `RadioButton`s if another button is checked, so we don't have to manually handle unchecking for every button.

One way we can handle it is like this
```kt
val id = when (medicineFrequency) {
  MedicineFrequencies.OD -> R.id.medicineFrequencyOdRadioButton
  MedicineFrequencies.BD -> R.id.medicineFrequencyBdRadioButton
  MedicineFrequencies.TDS -> R.id.medicineFrequencyTdsRadioButton
  MedicineFrequencies.QDS -> R.id.medicineFrequencyQdsRadioButton
}

medicineFrequencyRadioButton.check(id)
```

Another approach is having view id's inside the `MedicineFrequency` enum.